contracts/verifier.sol

```diff
@@ -0,0 +1,260 @@
+pragma solidity ^0.4.0;
+import {ClaimManager} from "./claimManager.sol";
+// Simple generic challenge-response computation verifier.
+//
+// @TODO:
+// * Multiple challangers (proposer should not win just because one challenger fails)
+// * Require "gas available" proof for timeout
+/**
+  * @title
+  * @author Christian Reitwiessner
+*/
+contract Verifier {
+
+    event NewClaim(uint sessionId);
+    event NewQuery(uint sessionId);
+    event NewResponse(uint sessionId);
+    event ChallengerConvicted(uint sessionId);
+    event ClaimantConvicted(uint sessionId);
+
+    uint constant responseTime = 1 hours;
+
+    struct VerificationSession {
+        uint id;
+        address claimant;
+        address challenger;
+        bytes input;
+        bytes output;
+        uint lastClaimantMessage;
+        uint lastChallengerMessage;
+        uint lowStep;
+        bytes32 lowHash;
+        uint medStep;
+        bytes32 medHash;
+        uint highStep;
+        bytes32 highHash;
+    }
+    VerificationSession[] public sessions;
+
+    function claimComputation(
+        address challenger,
+        address claimant,
+        bytes _input,
+        bytes _output,
+        uint steps
+    )
+        public
+        returns (uint)
+    {
+        require(steps > 2);
+
+        uint sessionId = sessions.length;
+        sessions.push(VerificationSession({
+            id: sessionId,
+            claimant: claimant,
+            challenger: challenger,
+            input: _input,
+            output: _output,
+            lastClaimantMessage: now,
+            lastChallengerMessage: now,
+            lowStep: 0,
+            lowHash: keccak256(_input),
+            medStep: 0,
+            medHash: 0,
+            highStep: steps,
+            highHash: keccak256(_output)
+        }));
+
+        require(isInitiallyValid(sessions[sessionId]));
+
+        NewClaim(sessionId);
+        return sessionId;
+    }
+
+    modifier onlyClaimant(uint sessionId) {
+        require(msg.sender == sessions[sessionId].claimant);
+        _;
+    }
+
+    // @TODO(shrugs) - this allows anyone to challenge an empty claim
+    //  is this what we want?
+    modifier onlyChallenger(uint sessionId) {
+        var session = sessions[sessionId];
+        if (session.challenger == address(0)) {
```

Comment:
This can be removed.

Reply:
Fixed in 94283548d2c23e2ccce29da2e60808767a218060